Tests/IndexerTest.php

```diff
@@ -401,4 +404,210 @@ public function ifUrlIsNotIndexedReturnTrue()
 
         $this->assertTrue($actual);
     }
+
+    public function testGetHashSolarId()
+    {
+        $url = 'https://github.com';
+        $hashIdUrl = sha1(strtolower(UrlCheck::fixUrl($url)));
+
+        $uriObj = new Uri('https://github.com');
+        $hashIdUri = sha1(strtolower(UrlCheck::fixUrl($uriObj->toString())));
+
+        $solrClient = $this->getMockBuilder('Solarium\Client')
+            ->setConstructorArgs([])
+            ->setMethods(['createSelect', 'select'])
+            ->getMock();
+
+        $indexer = new Indexer($solrClient, [], 50);
+
+        $this->assertEquals($indexer->getHashSolarId($url), $hashIdUrl);
+        $this->assertEquals($indexer->getHashSolarId($uriObj), $hashIdUri);
+        $this->assertEquals($hashIdUrl, $hashIdUri);
+    }
+
+    public function storedAndNotExpiredDataProvider()
+    {
+        $url1 = 'http://www.simgroep.nl/internet/medewerkers_41499/';
+        $url2 = 'http://www.simgroep.nl/internet/nieuws-uit-de-branche_41509/';
+        $url3 = 'http://www.simgroep.nl/internet/portfolio_41515/search/';
+        $url4 = 'http://www.simgroep.nl/internet/vacatures_41521/';
+
+        return [
+            [
+                [
+                    new Uri($url1),
+                    new Uri($url2),
+                    new Uri($url3),
+                    new Uri($url4)
+                ],
+                [
+                    sha1(strtolower(UrlCheck::fixUrl($url1))),
+                    sha1(strtolower(UrlCheck::fixUrl($url2))),
+                    sha1(strtolower(UrlCheck::fixUrl($url3))),
+                    sha1(strtolower(UrlCheck::fixUrl($url4)))
+                ]
+            ]
+        ];
+    }
+
+    /**
+     * @dataProvider storedAndNotExpiredDataProvider
+     */
+    public function testGetUnstoredOrExpiredUris($uris, $storedIds)
+    {
+        $urlUnstored = [new Uri('https://github.com')];
+
+        $solrClient = $this->getMockBuilder('Solarium\Client')
+            ->setConstructorArgs([])
+            ->getMock();
+
+        $indexer = new Indexer($solrClient, [], 50);
+
+        $result = $indexer->getUnstoredOrExpiredUris($urlUnstored, $storedIds);
+        $this->assertArrayHasKey(sha1(strtolower(UrlCheck::fixUrl($urlUnstored[0]))), $result);
+
+        $result = $indexer->getUnstoredOrExpiredUris($uris, $storedIds);
+        $this->assertEmpty($result);
+    }
+
+    public function urisNotUniqueDataProvider()
+    {
+        $url1 = new Uri('http://www.simgroep.nl/internet/medewerkers_41499/');
```

Comment:
do we really need this variables $url1...4 ? Can you return them in array without using temp variables?